### ISSUE TITLE
Cross compile errors (see below please)

### ISSUE THREAD

**Issue by SenseiHugeD** [vathpela/gnu-efi#17]
![Screenshot from 2023-04-12 18-03-16](https://user-images.githubusercontent.com/114719839/231619393-7f83b550-d408-4208-bbb2-147169fb790b.png)
So upon trying to cross compile the sources there appears to be some sort of issue here with missing functions. Unless I'm not doing something correctly this kind of sucks because I refuse to learn the Intel/AMD platforms just to get a working EFI bootloader made

**Comment by milkylainen** [comment]
Yep. Having the same errors for v7a arm cross-builds here with .17
iirc I don't think I had these errors when using .15. Probably anecdotal though.
Toolchain is a rather modern 12.2.0 with 2.37 glibc, ct-ng based.

**Comment by milkylainen** [comment]
After looking into it I think the compile error is because gnu-efi doesn't carry the needed set of ARM __aeabi* implementations with regard to floating point. It is freestanding after all. Anyway.. print.c uses floats and needs them. So either gnu-efi needs to implement them or you need softfp or hard as -mfloat-abi. A pure soft implementation won't currently do iiuc.

**Comment by syscall7788** [comment]
> ![Screenshot from 2023-04-12 18-03-16](https://user-images.githubusercontent.com/114719839/231619393-7f83b550-d408-4208-bbb2-147169fb790b.png) So upon trying to cross compile the sources there appears to be some sort of issue here with missing functions. Unless I'm not doing something correctly this kind of sucks because I refuse to learn the Intel/AMD platforms just to get a working EFI bootloader made

You need the arm-linux-gnueabihf compiler for this. Using this particular one and a fresh `make clean` fixes this
